WebSurveyApp/WebSurveyApp/Controllers/HomeController.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace WebSurveyApp.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly SurveyDbContext _context;
+
+        public HomeController(SurveyDbContext context)
+        {
+            _context = context;
+        }
+
+        [Authorize]
+        public async Task<IActionResult> List()
+        {
+            var surveys = await _context.Surveys
```

Comment:
please move all code not related to handling http requests out from controllers
actions should be similar to this:
```
public async Task<IActionResult> List(ListSurveysBindingModel model)
{
    var request = model.ToServiceModel();
    var surveys = await _someService.GetSurveysAsync(request);
    var viewModel = surveys.ToViewModel();
    return View(viewModel);
}
```